centreon/www/front_src/src/Resources/Actions/Resource/Check/helpers/index.ts

```diff
@@ -0,0 +1,29 @@
+import { equals } from 'ramda';
+
+import { Resource, ResourceCategory, ResourceType } from '../../../../models';
+
+interface CheckedResources {
+  resources: Array<Resource>;
+}
+
+interface Parent {
+  id: number;
+}
+
+interface PayloadCheckedResource {
+  id?: number;
+  parent: Parent | null;
+  type: ResourceCategory;
+}
+
+export const adjustCheckedResources = ({
+  resources
+}: CheckedResources): Array<PayloadCheckedResource> => {
+  const payload = resources.map(({ type, id, parent, service_id }) => ({
+    id: equals(type, ResourceType.anomalyDetection) ? service_id : id,
+    parent: parent ? { id: parent?.id } : null,
```

Comment:
if we only keep the parent Id, why not just  do 
```suggestion
    parentId: parent ? parent?.id : null,
```

Reply:
this is the form of the payload that the should sent to the backend

Reply:
ok, my bad.